op-node/rollup/derive/attributes.go

```diff
@@ -149,3 +158,39 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 func (ba *FetchingAttributesBuilder) CachePayloadByHash(payload *eth.ExecutionPayload) bool {
 	return ba.l2.CachePayloadByHash(payload)
 }
+
+func SnowL1GasPrice(ctx context.Context, ba *FetchingAttributesBuilder, epoch eth.BlockID) (*big.Int, error) {
+	// Consider this situation. If start a new l2 chain, starting from the block height of l1 less than CountBlockSize,
+	// in fact, this situation is unlikely to happen.
+	if epoch.Number < bsc.CountBlockSize-1 {
+		return bsc.DefaultBaseFee, nil
+	}
+	if latestBlockHash == epoch.Hash {
+		return latestL1GasPrice, nil
+	}
+	var allMedianGasPrice []*big.Int
+	blockHash := epoch.Hash
+	for len(allMedianGasPrice) < bsc.CountBlockSize {
+		if blockInfo, ok := bsc.BlockInfoCache.Get(blockHash); ok {
+			allMedianGasPrice = append(allMedianGasPrice, blockInfo.MedianGasPrice)
+			blockHash = blockInfo.ParentHash
+		} else {
+			block, transactions, err := ba.l1.InfoAndTxsByHash(ctx, blockHash)
+			if err != nil {
+				return nil, NewTemporaryError(fmt.Errorf("failed to fetch L1 block info and txs: %w", err))
```

Comment:
Since there is external requests here, what if the network is unstable and it returns error here? Will op-node stop, or there is higher level error handle logics to recover this?

Reply:
Return error  has no effect. The outermost layer of the entire block generation process is a loop, and this logic will also be entered. In fact, the call of this method `InfoAndTxsByHash` has been called before and put into the cache, in theory, it is taken directly from the cache. L1 will not actually be called.